### PR TITLE
Added GitHub Actions for iOS with TestFlight upload support

### PR DESCRIPTION
**Pull request by daniil-shumko**
# Description

NOTE: Android build is not yet working and hence is disabled!

Fixes #<issue_number>

# Checklist:

## Creator

- [x] The target branch is dev
- [ ] I have updated the unreleased section of the change log
- [x] I have reviewed the 'files changed' tab on github to ensure all changes are as expected
- [x] I have added someone to review the pr

## Reviewer

- [x] I have verified the above are all completed
- [ ] I have run the code locally to ensure it fufills the requirements of the issue
- [ ] I have reviewed the 'files changed' and commented on any sections which I think are not needed, incorrect or could be improved

## After pull

- [ ] If appropriate I have closed the issue

### CODE REVIEW

**Comment by daniil-shumko** [comment]
Before merging I think we must disable the triggers for this workflow so that we can only manually trigger builds

File: .github/workflows/flutter-ci.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
+name: Flutter CI
+
+# This workflow is triggered on pushes to the repository.
+
+on:
+  push:
+    branches: [dev]
+  pull_request: 
+    branches: [dev]
```
Comment (JElgar) [inline .github/workflows/flutter-ci.yml:9]: Can we add in staging and master to this? I'm also thinking we wont want this to run on pr once we merge it in
```suggestion
  push:
    branches: [dev, staging, master]
```

**Review by JElgar — COMMENTED** [review]
Yeah, good point @daniil-shumko why don't we only do it on pushes to staging or master as those should only happen when we want a release and then we can manually trigger dev builds if we want?

**Review by JElgar — APPROVED** [review]
Other than the trigger this looks awesome thank you @daniil-shumko

File: .github/workflows/flutter-ci.yml (hunk 1 above)
Comment (daniil-shumko) [reply .github/workflows/flutter-ci.yml:9]: Do staging builds need to be built differently?

Comment (JElgar) [reply .github/workflows/flutter-ci.yml:9]: They used to be but for now let's keep it simple

Comment (JElgar) [reply .github/workflows/flutter-ci.yml:9]: I think its probably better if we just keep them the same

**Pull request merged**